src/scalper.ts

```diff
@@ -424,6 +432,8 @@ export class Scalper {
     }
 
     // Place Gamma scalp bid & offer
+    let bidAttempts = 0;
+    while (bidAttempts < 3) {
```

Comment:
is there any cleanup of failed attempts?
Does cancel all orders do a cleanup?

Reply:
Cancel All orders each run cleans up excess this is to fix the issue where only 1 side of the gamma order goes thru. We always want both sides live

Reply:
We should avoid just retrying without backoff unless we know why it failed and believe that it will succeed on the next try.
Do we need a new bid id and cancel the failed one explicitly?

Reply:
If it is a transient error, we want to avoid a situation where we spam the endpoint and get throttled even though the requests are valid. So either backoff or have good reason to believe the next one succeeds, even though it is limited number of attempts here.

Reply:
Can't recall the specific error but observed this last night. Just the ask transaction made it on a block. Reason for limiting it to 3 was because a repeat tx would have very likely been confirmed as the order placed after the first was the failed one. Shouldn't need to refresh the gamma bid/ask id's since this only triggers when they don't make it to the order book

Reply:
Would a repeat tx be confirmed with the same id?

Reply:
Order IDs are optional on mango-client placeperporder2 and can repeat

Reply:
Alright for now, but should add a TODO to fix this logic. Generally it is bad practice to just immediately send a retry without better monitoring for whether the errors are transient or backoff of some sort.